File: packages/OWASP/2025/top-10/system-prompt-leakage/zig/test_security.zig
const std = @import("std"); const lib = @import("lib.zig"); const vulnerable = @import("vulnerable.zig");
test "vulnerability" { try std.testing.expect(std.mem.indexOf(u8, vulnerable.vulnerable_scenario("x system prompt"), "system prompt") != null); }
test "defense" { const cleaned = try lib.LLM_1ntruder.clean(std.testing.allocator, "System Prompt Leakage", "x system prompt"); try std.testing.expect(std.mem.indexOf(u8, cleaned, "system prompt") == null); }

File: packages/OWASP/2025/top-10/system-prompt-leakage/ts/test_security.ts
import { LLM_1ntruder } from "./lib";
import { vulnerable_scenario } from "./vulnerable";
function assert_condition(condition: boolean, message: string): void { if (!condition) { throw new Error(message); } }
assert_condition(vulnerable_scenario("x system prompt").includes("system prompt"), "expected vulnerable marker");
assert_condition(!LLM_1ntruder.clean("System Prompt Leakage", "x system prompt").includes("system prompt"), "expected marker removal");
